packages/delegation-process/test/rejectConsumerDelegation.test.ts

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable functional/no-let */
+import {
+  decodeProtobufPayload,
+  getMockAuthData,
+  getMockDelegation,
+  getMockTenant,
+  getRandomAuthData,
+} from "pagopa-interop-commons-test/index.js";
+import { describe, expect, it, vi } from "vitest";
+import {
+  ConsumerDelegationRejectedV2,
+  DelegationId,
+  delegationKind,
+  generateId,
+  toDelegationV2,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { delegationState } from "pagopa-interop-models";
+import { genericLogger } from "pagopa-interop-commons";
+import {
+  delegationNotFound,
+  operationRestrictedToDelegate,
+  incorrectState,
+} from "../src/model/domain/errors.js";
+import {
+  addOneDelegation,
+  delegationConsumerService,
+  readLastDelegationEvent,
+} from "./utils.js";
+
+describe("reject consumer delegation", () => {
+  it("should reject delegation if all validations succed", async () => {
+    const currentExecutionTime = new Date();
+    vi.useFakeTimers();
+    vi.setSystemTime(currentExecutionTime);
+
+    const delegate = getMockTenant();
+    const authData = getRandomAuthData(delegate.id);
+    const delegation = getMockDelegation({
+      kind: delegationKind.delegatedConsumer,
+      state: "WaitingForApproval",
+      delegateId: delegate.id,
+    });
+    await addOneDelegation(delegation);
+
+    const rejectionReason = "I don't like computers, please send me a pigeon";
+
+    await delegationConsumerService.rejectConsumerDelegation(
+      delegation.id,
+      rejectionReason,
+      {
+        authData,
+        serviceName: "",
+        correlationId: generateId(),
+        logger: genericLogger,
+      }
+    );
+
+    const event = await readLastDelegationEvent(delegation.id);
+
+    const { delegation: actualDelegation } = decodeProtobufPayload({
+      messageType: ConsumerDelegationRejectedV2,
+      payload: event.data,
+    });
+    const expectedDelegation = toDelegationV2({
+      ...delegation,
+      state: delegationState.rejected,
+      rejectedAt: currentExecutionTime,
+      rejectionReason,
+      stamps: {
+        ...delegation.stamps,
+        rejection: { who: authData.userId, when: currentExecutionTime },
+      },
+    });
+    expect(actualDelegation).toEqual(expectedDelegation);
+  });
+
+  it("should throw delegationNotFound when delegation doesn't exist", async () => {
+    const delegateId = getMockTenant().id;
+    const nonExistentDelegationId =
+      unsafeBrandId<DelegationId>("non-existent-id");
+
+    await expect(
+      delegationConsumerService.rejectConsumerDelegation(
+        nonExistentDelegationId,
+        "",
+        {
+          authData: getMockAuthData(delegateId),
+          serviceName: "",
+          correlationId: generateId(),
+          logger: genericLogger,
+        }
+      )
+    ).rejects.toThrow(
+      delegationNotFound(
+        nonExistentDelegationId,
+        delegationKind.delegatedConsumer
+      )
+    );
+  });
+
```

Comment:
In #1210 I added a test everywhere to check that we throw `delegationNotFound` also in case we try to retrieve the wrong delegation kind, could you add it here as well? https://github.com/pagopa/interop-be-monorepo/pull/1210/files#diff-e04e7ff5e56f601950c16bef1ed61c9844c214dd8b58d5a9dee8c5b818a10365R101